src/main/webapp/app/iris/base-chatbot/iris-base-chatbot.component.ts

```diff
@@ -0,0 +1,371 @@
+import { ButtonType } from 'app/shared/components/button.component';
+import { TranslateService } from '@ngx-translate/core';
+import { IrisLogoSize } from 'app/iris/iris-logo/iris-logo.component';
+import { IrisStageDTO, IrisStageStateDTO } from 'app/entities/iris/iris-stage-dto.model';
+import { IrisRateLimitInformation } from 'app/entities/iris/iris-ratelimit-info.model';
+import { IrisStatusService } from 'app/iris/iris-status.service';
+import { IrisMessageContentType, IrisTextMessageContent } from 'app/entities/iris/iris-content-type.model';
+import { AccountService } from 'app/core/auth/account.service';
+import { animate, group, style, transition, trigger } from '@angular/animations';
+import { IrisChatService } from 'app/iris/iris-chat-base.service';
+
+@Component({
+    selector: 'jhi-iris-base-chatbot',
+    templateUrl: './iris-base-chatbot.component.html',
+    styleUrls: ['./iris-base-chatbot.component.scss'],
+    animations: [
+        trigger('messageAnimation', [
+            transition(':enter', [
+                style({
+                    height: '0',
+                    transform: 'scale(0)',
+                }),
+                group([
+                    animate(
+                        '0.3s ease-in-out',
+                        style({
+                            height: '*',
+                        }),
+                    ),
+                    animate(
+                        '0.3s 0.1s cubic-bezier(.2,1.22,.64,1)',
+                        style({
+                            transform: 'scale(1)',
+                        }),
+                    ),
+                ]),
+            ]),
+        ]),
+    ],
+})
+export class IrisBaseChatbotComponent implements OnInit, OnDestroy, AfterViewInit {
+    // Icons
+    faTrash = faTrash;
+    faCircle = faCircle;
+    faPaperPlane = faPaperPlane;
+    faExpand = faExpand;
+    faXmark = faXmark;
+    faArrowDown = faArrowDown;
+    faCircleInfo = faCircleInfo;
+    faCompress = faCompress;
+    faThumbsUp = faThumbsUp;
+    faThumbsDown = faThumbsDown;
+    faRedo = faRedo;
+
+    protected readonly IrisSender = IrisSender;
+
+    // State variables
+    messagesSubscription: Subscription;
+    stagesSubscription: Subscription;
+    errorSubscription: Subscription;
+    numNewMessageSubscription: Subscription;
+    rateLimitSubscription: Subscription;
+    activeStatusSubscription: Subscription;
+
+    messages: IrisMessage[] = [];
+    stages?: IrisStageDTO[] = [];
+    error?: IrisErrorMessageKey;
+    numNewMessages: number = 0;
+    rateLimitInfo: IrisRateLimitInformation;
+    active: boolean = true;
+
+    newMessageTextContent = '';
+    isLoading: boolean;
+    shouldAnimate: boolean = false;
+    hasActiveStage: boolean = false;
+
+    // User preferences
+    userAccepted: boolean;
+    isScrolledToBottom = true;
+    rows = 1;
+    resendAnimationActive: boolean;
+    public ButtonType = ButtonType;
+
+    @Input() chatService: IrisChatService;
```

Comment:
please inject this service.

Reply:
I tried that (by somehow tricking the [DI Hierachy](https://v17.angular.io/guide/hierarchical-dependency-injection) to do it), but I couldn't make it work since only subclasses of `IrisChatService` are actually instantiated: one for the exercise chat, one for the course chat. The base component is shared between both features, and therefore we need to pass the correct service in. If you have a working solution that uses DI to solve this, lmk

Reply:
This is outdated